geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/publishAndSubscribe/Subscriptions.java

```diff
@@ -73,28 +73,52 @@ boolean exists(Object channelOrPattern, Client client) {
         .collect(Collectors.toList());
   }
 
-  public List<byte[]> findChannelNames() {
+  public List<Object> findChannelNames() {
 
     ObjectOpenCustomHashSet<byte[]> hashSet =
         new ObjectOpenCustomHashSet<>(ByteArrays.HASH_STRATEGY);
 
-    findChannels()
+    findChannelSubscriptions()
         .forEach(channel -> hashSet.add(channel.getSubscriptionName()));
 
     return new ArrayList<>(hashSet);
   }
 
-  public List<byte[]> findChannelNames(byte[] pattern) {
+  public List<Object> findChannelNames(byte[] pattern) {
 
     GlobPattern globPattern = new GlobPattern(Coder.bytesToString(pattern));
 
     return findChannelNames()
         .stream()
-        .filter(name -> globPattern.matches(Coder.bytesToString(name)))
+        .filter(name -> globPattern.matches(Coder.bytesToString((byte[]) name)))
         .collect(Collectors.toList());
   }
 
-  private Stream<ChannelSubscription> findChannels() {
+  public List<Object> findNumberOfSubscribersForChannel(List<byte[]> names) {
+
+    List<Object> result = new ArrayList<>();
+
+    names.forEach(name -> {
+
+      List<Subscription> subscriptions =
+          findSubscriptions(name)
+              .stream()
+              .filter(subscription -> subscription instanceof ChannelSubscription)
+              .collect(Collectors.toList());
+
+      if (subscriptions.isEmpty()) {
+        result.add(name);
+        result.add(0L);
+      } else {
+        result.add(name);
+        result.add((long) subscriptions.size());
+      }
```

Comment:
This can be simplified to:
```
      Long subscriptions =
          findSubscriptions(name)
              .stream()
              .filter(subscription -> subscription instanceof ChannelSubscription)
              .count();

        result.add(name);
        result.add(subscriptions);
```

Reply:
thank you-  yes, simpler!